test/e2e/pv_reclaimpolicy.go

```diff
@@ -0,0 +1,203 @@
+		var (
+			volumePath    string
+			pv            *v1.PersistentVolume
+			pvc           *v1.PersistentVolumeClaim
+			volumeoptions vsphere.VolumeOptions
+		)
+
+		It("should retain persistent volume when reclaimPolicy set to retain when associated claim is deleted", func() {
+
+			By("creating vmdk")
+			vsp, err := vsphere.GetVSphere()
+			Expect(err).NotTo(HaveOccurred())
+
+			volumeoptions.CapacityKB = 2097152
+			volumeoptions.Name = "e2e-disk" + time.Now().Format("20060102150405")
+			volumeoptions.DiskFormat = "thin"
+			volumePath, err = vsp.CreateVolume(&volumeoptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating the pv")
+			pv = getVspherePersistentVolumeSpec(volumePath, v1.PersistentVolumeReclaimRetain, nil)
+			pv, err := c.CoreV1().PersistentVolumes().Create(pv)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating the pvc")
+			pvc = getvSpherePersistentVolumeClaimSpec(ns, nil)
+
+			pvc, err := c.CoreV1().PersistentVolumeClaims(ns).Create(pvc)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("wait for the pv and pvc")
+			waitOnPVandPVC(c, ns, pv, pvc)
+
+			By("delete pvc")
+			deletePersistentVolumeClaim(c, pvc.Name, ns)
+
+			By("verify pv is retained")
+			framework.Logf("Waiting for PV %v to become Released", pv.Name)
+			err = framework.WaitForPersistentVolumePhase(v1.VolumeReleased, c, pv.Name, 3*time.Second, 300*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("delete pv")
+			deletePersistentVolume(c, pv.Name)
+			err = framework.WaitForPersistentVolumeDeleted(c, pv.Name, 3*time.Second, 300*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+})
+
+func getVspherePersistentVolumeSpec(volumePath string, persistentVolumeReclaimPolicy v1.PersistentVolumeReclaimPolicy, labels map[string]string) (*v1.PersistentVolume) {
+	var (
+		pvConfig persistentVolumeConfig
+		pv       *v1.PersistentVolume
+		claimRef *v1.ObjectReference
+	)
+	pvConfig = persistentVolumeConfig{
+		namePrefix: "vspherepv-",
+		pvSource: v1.PersistentVolumeSource{
+			VsphereVolume: &v1.VsphereVirtualDiskVolumeSource{
+				VolumePath:            volumePath,
+				FSType:                "ext4",
+			},
+		},
+		prebind: nil,
+	}
+
+	pv = &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: pvConfig.namePrefix,
+			Annotations: map[string]string{
+				volumehelper.VolumeGidAnnotationKey: "777",
+			},
+		},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeReclaimPolicy: persistentVolumeReclaimPolicy,
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceStorage): resource.MustParse("2Gi"),
+			},
+			PersistentVolumeSource: pvConfig.pvSource,
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadWriteOnce,
+			},
+			ClaimRef: claimRef,
+		},
+	}
+	if(labels!=nil) {
+		pv.Labels = labels;
+	}
+	return pv
+}
+
+func getvSpherePersistentVolumeClaimSpec(namespace string, labels map[string]string) (*v1.PersistentVolumeClaim) {
```

Comment:
Make the "vSphere" format consistent across all functions. The "V" in some functions is lower case and upper case in other functions.